new/prusti-specs/src/specifications/preparser.rs

```diff
@@ -0,0 +1,605 @@
+        }
+        false
+    }
+    /// Check if the input starts with the operator.
+    fn peek_operator(&self, operator: &str) -> bool {
+        for (i, c) in operator.char_indices() {
+            if let Some(TokenTree::Punct(punct)) = self.tokens.get(i) {
+                if punct.as_char() != c {
+                    return false;
+                }
+                if i + 1 < operator.len() && punct.spacing() != Spacing::Joint {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+        true
+    }
+    /// Check whether the input starts with an operator.
+    fn peek_any_operator(&self) -> bool {
+        self.peek_operator("==>") || self.peek_operator("&&")
+    }
+    /// Check if the input starts with the operator and if yes consume it.
+    fn check_and_consume_operator(&mut self, operator: &str) -> bool {
+        if !self.peek_operator(operator) {
+            return false;
+        }
+
+        let mut span: Option<Span> = None;
+        for _ in operator.chars() {
+            self.pop();
+            if let Some(maybe_span) = span {
+                span = maybe_span.join(self.last_span);
+            }
+            else {
+                span = Some(self.last_span);
+            }
+        }
+        self.last_span = span.unwrap();
+        true
+    }
+    /// Check if we have a nested assertion here.
+    fn check_and_consume_parenthesized_block(&mut self) -> Option<Group> {
+        if let Some(TokenTree::Group(group)) = self.tokens.front() {
+            if group.delimiter() == Delimiter::Parenthesis {
+                if let Some(TokenTree::Group(group)) = self.pop() {
+                    return Some(group);
+                } else {
+                    unreachable!();
+                }
+            }
+        }
+        None
+    }
+
+    fn parse_identifier(&mut self) -> Option<Ident> {
+        if let Some(TokenTree::Ident(ident)) = self.tokens.front() {
+            if let Some(TokenTree::Ident(ident)) = self.pop() {
+                return Some(ident);
+            }
+        }
+        None
+    }
+
+    /// Creates a TokenStream until a certain operator is met, or until the end of the stream.
+    /// The terminating operator is not consumed.
+    fn create_stream_until(&mut self, operator: &str) -> TokenStream {
+        let mut stream = TokenStream::new();
+        let mut t = vec![];
+        while !self.peek_operator(operator) && !self.is_empty() {
+            t.push(self.pop().unwrap());
+        }
+        stream.extend(t.into_iter());
+        stream
+    }
+
+    /// Convert the content into TokenStream.
+    fn create_stream(&mut self) -> TokenStream {
+        let mut stream = TokenStream::new();
+        let mut t = vec![];
+        while !self.is_empty() {
+            t.push(self.pop().unwrap());
+        }
+        stream.extend(t.into_iter());
+        stream
+    }
+}
+
+#[derive(Debug)]
+struct ForAllArg {
```

Comment:
Why do you need both `Arg` and `ForAllArg`, which are basically identical?

Reply:
`ForAllArg` has that `:` token in it, which is a trash for further processing.

Reply:
So, just drop that `:` in the parsing function.